packages/api/src/middleware/validation/checkProductValidity.js

```diff
@@ -51,6 +62,15 @@ export function checkProductValidity() {
       ) {
         return res.status(400).send('percent elemental values must not exceed 100');
       }
+
+      // Check that compound values are all positive
+      if (!molecularCompounds.every((compound) => !sap[compound] || sap[compound] >= 0)) {
+        return res.status(400).send('compounds values must all be positive');
+      }
+
+      if (sap.moisture_content_percent && !sap.moisture_content_percent >= 0) {
+        return res.status(400).send('moisture content value must be positive');
```

Comment:
Did you also want to put the `<= 100` check on `moisture_content` here, or let the db handle it?

Reply:
I think I try to let the natural db errors handle it... to not have extra code.